front/lib/actions/mcp_internal_actions/servers/webtools_edge.ts

```diff
@@ -0,0 +1,335 @@
+  }
+  return new Ok(finalContent);
+}
+
+// Edge variant of webtools. Same tools and behavior as web_search_&_browse.
+const createServer = (
+  auth: Authenticator,
+  agentLoopContext?: AgentLoopContextType
+): McpServer => {
+  const server = makeInternalMCPServer("webtools_edge");
+
+  server.tool(
+    WEBSEARCH_TOOL_NAME,
+    "A tool that performs a Google web search based on a string query.",
+    {
+      query: z
+        .string()
+        .describe(
+          "The query used to perform the Google search. If requested by the " +
+            "user, use the Google syntax `site:` to restrict the search " +
+            "to a particular website or domain."
+        ),
+      page: z
+        .number()
+        .optional()
+        .describe(
+          "A 1-indexed page number used to paginate through the search results." +
+            " Should only be provided if the page is strictly greater than 1 in order" +
+            " to go deeper into the search results for a specific query."
+        ),
+    },
+    async ({ query, page }) => {
+      if (!agentLoopContext?.runContext) {
+        throw new Error(
+          "agentLoopRunContext is required where the tool is called."
+        );
+      }
+
+      const agentLoopRunContext = agentLoopContext.runContext;
+
+      const { websearchResultCount, citationsOffset } =
+        agentLoopRunContext.stepContext;
+
+      const websearchRes = await webSearch({
+        provider: "serpapi",
+        query,
+        page,
+        num: websearchResultCount,
+      });
+
+      if (websearchRes.isErr()) {
+        return makeMCPToolTextError(
+          `Failed to search: ${websearchRes.error.message}`
+        );
+      }
+
+      const refs = getRefs().slice(
+        citationsOffset,
+        citationsOffset + websearchResultCount
+      );
+
+      const results: WebsearchResultResourceType[] = [];
+      for (const result of websearchRes.value) {
+        results.push({
+          mimeType: INTERNAL_MIME_TYPES.TOOL_OUTPUT.WEBSEARCH_RESULT,
+          title: result.title,
+          text: result.snippet,
+          uri: result.link,
+          reference: refs.shift() as string,
+        });
+      }
+
+      return {
+        isError: false,
+        content: [
+          ...results.map((result) => ({
+            type: "resource" as const,
+            resource: result,
+          })),
+          {
+            type: "resource" as const,
+            resource: {
+              mimeType: INTERNAL_MIME_TYPES.TOOL_OUTPUT.WEBSEARCH_QUERY,
+              text: query,
+              uri: "",
+            },
+          },
+        ],
+      };
+    }
+  );
+
+  server.tool(
+    WEBBROWSER_TOOL_NAME,
+    "A tool to browse websites. Provide a list of URLs to browse. For each URL, the tool will return a summary of the content and a link to a file that contains the entire content.",
+    {
+      urls: z.string().array().describe("List of urls to browse"),
+      summaryAgent:
```

Comment:
Why is the summaryAgent a tool param? Could we keep it simple and  always use `GLOBAL_AGENTS_SID.DUST_BROWSER_SUMMARY`? 
(Being able to set your your summary agent may not be useful, so we could start simple)

How do you expect to roll this out? 
1/ We're confident this is much better always and this is just the new browse. 
2/ We roll it out by saying this is browse but with a "summary" option ticked. This would use our summary agent. 
3/ We roll it out by allowing to select a sub agent to summarize (without any guarantee the selected agent does this).

Reply:
I think this should be an option of the browse tool. You can set a "summary agent".

Maybe you are right and we should just have box to tick and alway use the default summary agent ?

Reply:
For sure this won't replace the webtools: most  use cases you don't want any summary, it just adds latency.

Let's do 2/ then